docs/docs/examples/guides/05-jtc-guide.md

```diff
@@ -0,0 +1,602 @@
+one. That is, **there is no trajectory buffering or appending taking place**. As with many things in the AICA Universe,
+behaviors are event-driven. If you wish to send multiple trajectories back-to-back, you will have to rely on the
+execution status of the active trajectory handled by JTC. There is a practical example of how do this in following
+sections (see [Putting an application together](#putting-an-application-together)). 
+
+### Trajectory execution status
+
+The controller exposes 4 predicates to reflect the the execution status of a trajectory, namely:
+
+- `Has active trajectory`: A trajectory has been set and is being executed
+- `Has trajectory succeeded`: A trajectory was executed successfully (i.e., reached the final waypoint respecting all 
+tolerances in the way)
+- `Has trajectory failed`: A trajectory failed to execute because a tolerance was violated (i.e., desired waypoint was
+not reached and/or time duration was exceeded)
+- `Is trajectory cancelled`: A user-triggered request to cancel the trajectory was successfully processed
+
+#### Using JTC with signals
+
+In practice, this requires publishing a single
+[`JointTrajectory`](https://docs.ros.org/en/noetic/api/trajectory_msgs/html/msg/JointTrajectory.html) message. In AICA
+Studio terms, that means connecting an edge from a custom component directly to the input of JTC. If you already have a
+component that has the corresponding signal output, feel free to skip the next segment.
+
+If you are writing the signal logic yourself, the following code indicates what your message-related code may look like:
+
+<Tabs groupId="programming-language">
+<TabItem value="python" label="Python">
+
+```python title="py_component.py"
+import rclpy
+from trajectory_msgs.msg import JointTrajectory, JointTrajectoryPoint
+
+# Typically in your constructor
+self._trajectory = JointTrajectory()
+self.add_output("trajectory", "_trajectory", JointTrajectory, publish_on_step=False)
+
+# Typically in a conditional block that will write the output upon the success of some criteria
+self._trajectory.header.stamp = self.get_clock().now().to_msg()
+self._trajectory.joint_names = ['joint_0', 'joint_1', 'joint_2', 'joint_3', 'joint_4', 'joint_5']
+
+p1 = JointTrajectoryPoint()
+p1.positions = [0.6, 0.2, 0.0, 0.0, -0.2, 0.0]
+p1.velocities = [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+p1.time_from_start = rclpy.time.Duration(seconds=2.0).to_msg()
+
+p2 = JointTrajectoryPoint()
+p2.positions = [-0.6, -0.2, 0.0, 0.0, 0.2, 0.0]
+p2.velocities = [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+p2.time_from_start = rclpy.time.Duration(seconds=4.0).to_msg()
+
+self._trajectory.points = [p1, p2]
+
+self.publish_output("trajectory")
+```
+
+</TabItem>
+<TabItem value="c++" label="C++">
+
+```cpp title="include/custom_component_package/CppComponent.hpp"
+#pragma once
+
+#include <trajectory_msgs/msg/joint_trajectory.hpp>
+#include <trajectory_msgs/msg/joint_trajectory_point.hpp>
+
+// ... 
```

Comment:
probably meant to finish a relevant comment here about declaring the class data member